portal-impl/src/com/liferay/portlet/documentlibrary/service/impl/DLFileEntryLocalServiceImpl.java

```diff
@@ -2771,6 +2777,80 @@ private long _getValidFileEntryTypeId(
 		}
 	}
 
+	private void _inheritRolesPermissions(
+			long inheritableParentFolderId, DLFileEntry dlFileEntry)
+		throws PortalException {
+
+		long companyId = dlFileEntry.getCompanyId();
+		String parentClassName = DLFolderConstants.getClassName();
+
+		if (inheritableParentFolderId ==
+				DLFolderConstants.DEFAULT_PARENT_FOLDER_ID) {
+
+			int count =
+				_resourcePermissionLocalService.getResourcePermissionsCount(
+					companyId, DLConstants.RESOURCE_NAME,
+					ResourceConstants.SCOPE_INDIVIDUAL,
+					String.valueOf(dlFileEntry.getGroupId()));
+
+			if (count == 0) {
+				_resourceLocalService.addResources(
+					companyId, dlFileEntry.getGroupId(), 0,
+					DLConstants.RESOURCE_NAME,
+					String.valueOf(dlFileEntry.getGroupId()), false, true,
+					true);
+			}
+
+			parentClassName = DLConstants.RESOURCE_NAME;
+			inheritableParentFolderId = dlFileEntry.getGroupId();
+		}
+
+		Map<Long, Set<String>> dlFolderRoleIdsToActionIds =
+			_resourcePermissionLocalService.
+				getAvailableResourcePermissionActionIds(
+					companyId, parentClassName,
+					ResourceConstants.SCOPE_INDIVIDUAL,
+					String.valueOf(inheritableParentFolderId),
+					ResourceActionsUtil.getModelResourceActions(
```

Comment:
get _commonpermission here

Reply:
Fixed it. Thanks

Reply:
It will cause problems if you do that. please try removing all common permissions in Folder Permissions and add a new document to test

Reply:
I tested it and there is no problem.

Reply:
Step reproduce:
1. Create a new folder with propagation = true;
2. Edit the newly created folder's permissions
3. Remove all common permissions -> save
4. Upload a document to the folder
5. Observe that the newly uploaded document still has common permissions.

Reply:
Sorry, I missed that case. 
Fixed it. Thanks